shop/src/main/java/ro/msg/learning/shop/service/ProductService.java

```diff
@@ -0,0 +1,85 @@
+package ro.msg.learning.shop.service;
+
+import org.hibernate.service.spi.ServiceException;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.DTO.ProductCategoryDTO;
+import ro.msg.learning.shop.model.Product;
+import ro.msg.learning.shop.repository.ProductCategoryRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.SupplierRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ProductService {
+    @Autowired
+    ProductRepository productRepository;
+    @Autowired
+    ProductCategoryRepository productCategoryRepository;
+    @Autowired
+    SupplierRepository supplierRepository;
+
+
+    private final ModelMapper modelMapper;
+
+    public ProductService(ProductRepository productRepository, ModelMapper modelMapper) {
+        this.productRepository = productRepository;
+        this.modelMapper = modelMapper;
+    }
+
+    public ProductCategoryDTO mapToDTO(Product product){
+        ProductCategoryDTO productCategoryDTO = modelMapper.map(product,ProductCategoryDTO.class);
+        return productCategoryDTO;
+    }
+
+    public Product mapToEntity(ProductCategoryDTO productCategoryDTO){
+        Product product = modelMapper.map(productCategoryDTO,Product.class);
+        return product;
+    }
+
+    public List<Product> getAllProducts() {
+        return productRepository.findAll();
+    }
+
+    public Optional<Product> getProductById(Integer id){
```

Comment:
The service methods should never return an Optional. Verify in the method if the object is present, otherwise throw an error.